apollo-api/src/main/java/com/apollographql/android/api/graphql/Field.java

```diff
@@ -4,72 +4,70 @@
 import java.util.Collections;
 import java.util.Map;
 
-/** TODO */
-public final class Field {
+/** TODO add javadocs */
+public class Field {
   private final Type type;
   private final String responseName;
   private final String fieldName;
   private final Map<String, Object> arguments;
-  private final ObjectReader objectReader;
-  private final ListReader listReader;
   private final boolean optional;
-  private final ScalarType scalarType;
 
   public static Field forString(String responseName, String fieldName, Map<String, Object> arguments,
       boolean optional) {
-    return new Field(Type.STRING, responseName, fieldName, arguments, null, null, optional, null);
+    return new Field(Type.STRING, responseName, fieldName, arguments, optional);
   }
 
   public static Field forInt(String responseName, String fieldName, Map<String, Object> arguments, boolean optional) {
-    return new Field(Type.INT, responseName, fieldName, arguments, null, null, optional, null);
+    return new Field(Type.INT, responseName, fieldName, arguments, optional);
   }
 
   public static <T> Field forLong(String responseName, String fieldName, Map<String, Object> arguments,
       boolean optional) {
-    return new Field(Type.LONG, responseName, fieldName, arguments, null, null, optional, null);
+    return new Field(Type.LONG, responseName, fieldName, arguments, optional);
   }
 
   public static Field forDouble(String responseName, String fieldName, Map<String, Object> arguments,
       boolean optional) {
-    return new Field(Type.DOUBLE, responseName, fieldName, arguments, null, null, optional, null);
+    return new Field(Type.DOUBLE, responseName, fieldName, arguments, optional);
   }
 
   public static Field forBoolean(String responseName, String fieldName, Map<String, Object> arguments,
       boolean optional) {
-    return new Field(Type.BOOLEAN, responseName, fieldName, arguments, null, null, optional, null);
+    return new Field(Type.BOOLEAN, responseName, fieldName, arguments, optional);
   }
 
   public static <T> Field forObject(String responseName, String fieldName, Map<String, Object> arguments,
       boolean optional, ObjectReader<T> objectReader) {
-    return new Field(Type.OBJECT, responseName, fieldName, arguments, objectReader, null, optional, null);
+    return new ObjectField(responseName, fieldName, arguments, optional, objectReader);
```

Comment:
I kinda find it weird that a method in the parent class returns an instance of one of it subclasses. (unless there's a use-case outside this PR changes)

Wouldn't it be better just to have those subclasses `ObjectField, ScalarListField, CustomTypeField` defined outside and have those methods define there instead?

In both cases we'll have to do the cast [here](https://github.com/apollographql/apollo-android/pull/157/files#diff-fcfdd9c152da5f099c8263ad8783940cR52) anyways.